modules/stack/main.tf

```diff
@@ -1,3 +1,26 @@
+locals {
+  triggers = [for trigger in var.triggers : "depends-on:${trigger}|state:FINISHED"]
+
+  component_stack_deps = [for component_stack_dep in var.component_stack_deps : "stack-deps:${component_stack_dep}"]
+
+  imports = [for import in var.imports : "import:${import}"]
+
+  stack_config_name_parts = split("-", var.stack_config_name)
+
+  folders = concat(
+    format("folder:%s", try(local.stack_config_name_parts[0], "")),
+    format("folder:%s/%s", try(local.stack_config_name_parts[0], ""), try(local.stack_config_name_parts[1], "")),
+    format("folder:component/%s", var.logical_component)
```

Comment:
Is this imposing a string naming convention on stack files?

Reply:
This should not be imposing any file naming convention. Folders in spacelift are like labels in Gmail. We can have as many as we want. These are the ones we are starting with and I think we should ignore the folder if the data is not available